File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        char[] chars = myString.toCharArray();
        for(int i = 0; i < chars.length; i++)
            chars[i] = chars[i] > 'A' ? 'A' : 'B';
        
        return new String(chars).contains(pat) ? 1 : 0;
    }
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string myString, string pat) {
    string pat_target = "";
    for(char c : pat){
        pat_target += (c == 'A' ?  'B' : 'A');
    }
    return myString.find(pat_target) == string::npos ? 0 : 1;
}